src/Server.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-swoole for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-swoole/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Swoole;
+
+use Swoole\Http\Server as SwooleHttpServer;
+
+class Server
+{
+
+    /**
+     * @var string
+     */
+    protected $host;
+
+    /**
+     * @var int
+     */
+    protected $port;
+
+    /**
+     * @var int
+     */
+    protected $mode;
+
+    /**
+     * @var int
+     */
+    protected $protocol;
+
+    /**
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * @var SwooleHttpServer
+     */
+    protected $swooleServer;
+
+    public function __construct(string $host, int $port, int $mode, int $protocol, array $options)
```

Comment:
Please provide an `@see` annotation pointing users to the location in the Swoole documentation detailing what options are allowed.

Reply:
Also provide an `@see` annotation to the documentation detailing the valid `$mode` and `$protocol` values.

Ideally, we should provide defaults for `$mode` and `$protocol`, and a default `[]` value for `$options`.

Reply:
@weierophinney default value for `$mode` and `$protocol` have been setted by `ServerFactory`, If still init the default value here, need to modify twice when we want to change the default value of some property

Reply:
Also `$mode` and `protocol` is requires in `__construct` arguments, I think provide defaults for properties are unnecessary